### PR TITLE
Allows bootstrapping juju 3.0 to other arch's.

### PR DESCRIPTION
**Pull request by tlm**
Due to a recent change in juju where the controller is now considered a charm we were not passing along constraints to the application deployment of the charm. This resulted in a case where the controller charm was always trying to be deployed with amd64 instead of the arch specified at Bootstrap time.

Now these arch constraints are passed along to the controller charm deployment.

## Checklist

*If an item is not applicable, use `~strikethrough~`.*

- [x] Code style: imports ordered, good names, simple structure, etc
- [x] Comments saying why design decisions were made
- [x] Go unit tests, with comments saying what you're testing
- [x] [Integration tests](https://github.com/juju/juju/tree/develop/tests), with comments saying what you're testing
- [x] [doc.go](https://discourse.charmhub.io/t/readme-in-packages/451) added or updated in changed packages

## QA steps

To test this you will need to have a bootstrap target that isn't amd64. Can either set up a vm or arm64 based machine in AWS with LXD.

Bootstrap to the new machine with:

`juju bootstrap --bootstrap-constraints arch=arm64`

If everything is fixed the bootstrap should succeed with no errors. See the bug report for behavior pre this change.

## Documentation changes

N/A This should have worked out of the gate.

## Bug reference

https://bugs.launchpad.net/juju/+bug/1994173

### CODE REVIEW

File: cmd/jujud/agent/bootstrap_test.go [hunk 1]
```diff
@@ -199,6 +199,28 @@ func (s *BootstrapSuite) TestLocalControllerCharm(c *gc.C) {
...
+	c.Assert(err, jc.ErrorIsNil)
+	defer loggo.RemoveWriter("bootstrap-test")
+
+	err = cmd.Run(nil)
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(tw.Log(), jc.LogMatches, jc.SimpleMessages{{
+		loggo.DEBUG,
+		`Successfully deployed local Juju controller charm`,
+	}})
+	s.assertControllerApplication(c)
```
Comment (wallyworld) [inline cmd/jujud/agent/bootstrap_test.go:221]: We should check that the controller application is recorded in state with the expected constraints.

**Review by wallyworld — APPROVED** [review]
Just needs a small test imporvement

File: cmd/jujud/agent/bootstrap_test.go (hunk 1 above)
Comment (tlm) [reply cmd/jujud/agent/bootstrap_test.go:233]: Try that.

**Pull request merged**